Basic auth is not working

A simple config with basic auth configured leads to panic. There is no documentation regarding basic auth, but from the source it seems that it should work.

```
global:
  evaluation_interval: 5s

promxy:
  server_groups:
    - static_configs:
      - targets:
        - localhost/prometheus1
        - localhost/prometheus2
      anti_affinity: 10s
      http_client:
        basic_auth:
          username: 'user'
          password: 'password'
        tls_config:
          insecure_skip_verify: true
```
```
./promxy --config=promxy.yml
panic: interface conversion: http.RoundTripper is *config.basicAuthRoundTripper, not *http.Transport

goroutine 1 [running]:
github.com/jacksontj/promxy/servergroup.(*ServerGroup).ApplyConfig(0xc42029ad20, 0xc420362200, 0x1, 0xc4202048c8)
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/servergroup/servergroup.go:133 +0x40e
github.com/jacksontj/promxy/proxystorage.(*ProxyStorage).ApplyConfig(0xc420539b70, 0xc4204601c0, 0xc4204601c0, 0x0)
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/proxystorage/proxy.go:70 +0x146
main.reloadConfig(0xc420540180, 0x5, 0x8, 0x1d36266, 0x8)
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/cmd/promxy/main.go:84 +0xc4
main.main()
	/home/tjackson/workspace/golang/src/github.com/jacksontj/promxy/cmd/promxy/main.go:303 +0x13a7
```

--- a/servergroup/servergroup.go
+++ b/servergroup/servergroup.go
@@ -130,8 +130,14 @@ func (s *ServerGroup) ApplyConfig(cfg *Config) error {
 	// dial timeout (2) healthchecks (3) track "healthiness" of downstream based on our
 	// requests to it -- not through other healthchecks
 	// Override the dial timeout
-	transport := client.Transport.(*http.Transport)
-	transport.DialContext = (&net.Dialer{Timeout: 200 * time.Millisecond}).DialContext
+	switch transport := client.Transport.(type) {
+	case *http.Transport:
+		transport.DialContext = (&net.Dialer{Timeout: 200 * time.Millisecond}).DialContext
+		// TODO: basic auth?  as reported in #70 basicAuth doesn't use this timeout.
+		// This is because prometheus has its own RoundTripper (*config.basicAuthRoundTripper)
+		// which doesn't set a timeout or expose a way to do so. For now I'm changing
+		// this just so it won't panic, but this is not a long-term solution to the issue.
+	}
 
 	s.Client = client
 

